MIPs/mip-x-evmmultichainapi.md

```diff
@@ -0,0 +1,176 @@
+---
+mip: x
+title: Multichain API for Ethereum
+status: Draft
+stability: n/a
+discussions-to: https://github.com/MetaMask/metamask-improvement-proposals/discussions/53
+author(s): Alex Donesky (@adonesky1), Jiexi Luan (@jiexi), Vandan Parikh(@vandan) 
+type: Maintainer
+created: 2024-10-28
+---
+
+## Summary
+This proposal supplements [MIP-x](./mip-x-multichainapi.md) with additional specifications for EVM networks.
+
+## Motivation
+Current single chain EVM Wallet APIs severely limit EVM dapp development and cause a great deal of unnecessary friction for users operating in an ecosystem with rapidly growing number of chains. As such, Dapp developers and end users stand to benefit tremendously from the improved DevEx and UX that the Multichain API will provide. 
+
+The Multichain API will enable the following benefits for developers:
+- Concurrent cross chain state reading
+- Elimination of chain switching
+- Improved connection flows, with better interfaces for dapps to understand wallet capabilities and add missing networks on the fly.
+
+## Usage Example
+### Request
+An example structure for a JSON-RPC request that an application would send to request authorization(s) from MetaMask:
+
+```js
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "wallet_createSession",
+  "params": {
+    "optionalScopes": {
+      "wallet:eip155": {
+        "methods": ["wallet_addEthereumChain"],
+        "notifications": [],
+      },
+      "eip155:1": {
+        "methods": ["eth_sendTransaction","eth_call","eth_getBalance","eth_blockNumber","eth_getTransactionCount","wallet_watchAsset","eth_subscribe"],
+        "notifications": ["eth_subscription"],
+      },
+      "eip155:59144": {
+        "methods": ["eth_sendTransaction","eth_call","eth_getBalance","eth_blockNumber","eth_getTransactionCount","wallet_watchAsset","eth_subscribe"],
+        "notifications": ["eth_subscription"],
+      },
+    },
+    "scopedProperties": {
+      "eip155:1": {
+        "eip3085": {
+          "rpcEndpoints": [{
```

Comment:
do we need to nest it like this? Seems like it could be
"scopedProperties": {
      "eip155:1": {
        "eip3085": [
            "chainName": "Ethereum (Infura)",
            "rpcUrls": ["https://mainnet.infura.io"],
            "nativeCurrency": {
                "name": "ETH",
                "symbol": "ETH",
                "decimals": 18,
            },
       ]
    }
}

Reply:
Yeah, `rpcEndpoints` seemed superfluous. I've attempted to adjust the syntax according to that.